tests/components/lcn/test_climate.py

```diff
@@ -0,0 +1,360 @@
+"""Test for the LCN climate platform."""
+
+from unittest.mock import patch
+
+from pypck.inputs import ModStatusVar, Unknown
+from pypck.lcn_addr import LcnAddr
+from pypck.lcn_defs import Var, VarValue
+from syrupy.assertion import SnapshotAssertion
+
+# pylint: disable=hass-component-root-import
+from homeassistant.components.climate import DOMAIN as DOMAIN_CLIMATE
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACMode,
+)
+from homeassistant.components.lcn.helpers import get_device_connection
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import MockConfigEntry, MockModuleConnection, MockPchkConnectionManager
+
+
+async def test_setup_lcn_climate(
+    hass: HomeAssistant,
+    lcn_connection: MockPchkConnectionManager,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the setup of climate."""
+    state = hass.states.get("climate.climate1")
+    assert state is not None
+    assert state.state == snapshot(name=f"{state.entity_id}-state")
+
+
+async def test_entity_state(
+    hass: HomeAssistant,
+    lcn_connection: MockPchkConnectionManager,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test state of entity."""
+    state = hass.states.get("climate.climate1")
+    assert state is not None
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == snapshot(
+        name=f"{state.entity_id}-supported_features"
+    )
+    assert set(state.attributes[ATTR_HVAC_MODES]) == snapshot(
+        name=f"{state.entity_id}-hvac_modes"
+    )
+    assert state.attributes[ATTR_MIN_TEMP] == snapshot(
+        name=f"{state.entity_id}-min_temp"
+    )
+    assert state.attributes[ATTR_MAX_TEMP] == snapshot(
+        name=f"{state.entity_id}-max_temp"
+    )
+    assert state.attributes[ATTR_CURRENT_TEMPERATURE] == snapshot(
+        name=f"{state.entity_id}-current_temperature"
```

Comment:
Replace these tests with `snapshot_platform`, it's better to just capture the whole structure instead of single attributes

Reply:
Done